tests/2d_examples/test_2d_shell_collision/src/shell_collision.cpp

```diff
@@ -0,0 +1,277 @@
+ * @details This is the first case for test shell modeling.
+ * @author 	Massoud Rezavand and Xiangyu Hu
+ */
+#include "sphinxsys.h"	//SPHinXsys Library.
+using namespace SPH;	//Namespace cite here.
+//----------------------------------------------------------------------
+//	Basic geometry parameters and numerical setup.
+//----------------------------------------------------------------------
+Real DL = 4.0; 					/**< box length. */
+Real DH = 4.0; 					/**< box height. */
+Real resolution_ref = 0.025; 	/**< reference resolution. */
+Real BW = resolution_ref * 1; 	/**< wall width for BCs. */
+BoundingBox system_domain_bounds(Vec2d(-BW, -BW), Vec2d(DL + BW, DH + BW));
+Vec2d ball_center(2.0, 2.0);
+Real ball_radius = 0.5;			
+Real gravity_g = 1.0;
+//----------------------------------------------------------------------
+//	Global paramters on material properties
+//----------------------------------------------------------------------
+Real rho0_s = 1.0e3;
+Real Youngs_modulus = 5.0e4;
+Real poisson = 0.45; 			
+Real physical_viscosity = 10000.0; 
+//----------------------------------------------------------------------
+//	Bodies with cases-dependent geometries (ComplexShape).
+//----------------------------------------------------------------------
+class WallBoundary : public SolidBody
+{
+public:
+	WallBoundary(SPHSystem &sph_system, std::string body_name) : SolidBody(sph_system, body_name)
+	{
+		std::vector<Vecd> outer_wall_shape;
+		outer_wall_shape.push_back(Vecd(-BW, -BW));
+		outer_wall_shape.push_back(Vecd(-BW, DH + BW));
+		outer_wall_shape.push_back(Vecd(DL + BW, DH + BW));
+		outer_wall_shape.push_back(Vecd(DL + BW, -BW));
+		outer_wall_shape.push_back(Vecd(-BW, -BW));
+
+		std::vector<Vecd> inner_wall_shape;
+		inner_wall_shape.push_back(Vecd(0.0, 0.0));
+		inner_wall_shape.push_back(Vecd(0.0, DH));
+		inner_wall_shape.push_back(Vecd(DL, DH));
+		inner_wall_shape.push_back(Vecd(DL, 0.0));
+		inner_wall_shape.push_back(Vecd(0.0, 0.0));
+
+		body_shape_ = new ComplexShape(body_name);
+		body_shape_->addAPolygon(outer_wall_shape, ShapeBooleanOps::add);
+		body_shape_->addAPolygon(inner_wall_shape, ShapeBooleanOps::sub);
+	}
+};
+class FreeBall : public SolidBody
+{
+public:
+	FreeBall(SPHSystem& system, std::string body_name) : SolidBody(system, body_name)
+	{
+		ComplexShape original_body_shape;
+		original_body_shape.addACircle(ball_center, ball_radius, 100, ShapeBooleanOps::add);
+		body_shape_ = new LevelSetComplexShape(this, original_body_shape);
+	}
+};
+
+//----------------------------------------------------------------------
+//	Case dependent material properties
+//----------------------------------------------------------------------
+class WallMaterial : public LinearElasticSolid
+{
+public:
+	WallMaterial() : LinearElasticSolid()
+	{
+		rho0_ = rho0_s;
+		youngs_modulus_ = Youngs_modulus;
+		poisson_ratio_ = poisson;
+
+		assignDerivedMaterialParameters();
+	}
+};
+class BallMaterial : public NeoHookeanSolid
```

Comment:
same for the NeoHookean

Reply:
it is not relevant to the new changes of SHPinXsys